src/pretix/control/utils.py

```diff
@@ -0,0 +1,201 @@
+import logging
+from functools import wraps
+
+import stripe
+from django.core.exceptions import ValidationError
+
+from pretix.base.models import Organizer
+from pretix.base.models.organizer import OrganizerBillingModel
+from pretix.base.settings import GlobalSettingsObject
+
+logger = logging.getLogger(__name__)
+
+
+def get_stripe_key(key_type: str) -> str:
+    gs = GlobalSettingsObject()
+
+    try:
+        prod_key = getattr(gs.settings, "payment_stripe_connect_{}_key".format(key_type))
+        test_key = getattr(gs.settings, "payment_stripe_connect_test_{}_key".format(key_type))
+    except AttributeError as e:
+        raise ValidationError(
+            "Missing attribute for Stripe {} key: {}. Please contact the administrator to set the Stripe key.".format(
+                key_type, str(e)),
+        )
+
+    if not prod_key and not test_key:
+        raise ValidationError(
+            "Please contact the administrator to set the Stripe {} key.".format(key_type)
+        )
+
+    return prod_key or test_key
+
+
+def get_stripe_secret_key() -> str:
+    return get_stripe_key("secret")
+
+
+def get_stripe_publishable_key() -> str:
+    return get_stripe_key("publishable")
+
+
+stripe.api_key = get_stripe_secret_key()
+
+
+def handle_stripe_errors(operation_name: str):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except stripe.error.APIError as e:
+                logger.error("Stripe API error during %s: %s", operation_name, str(e))
+                raise ValidationError(
+                    "Stripe service error: {}".format(getattr(e, "user_message", str(e)))
+                )
+            except stripe.error.APIConnectionError as e:
+                logger.error(
+                    "API connection error during {}: %s", operation_name, str(e)
```

Comment:
ensuring the message is constructed only if the log level matches.
except stripe.error.APIConnectionError as e:
    logger.error(
        "API connection error during %s: %s", operation_name, str(e)
    )
    raise ValidationError("Network communication error: {}".format(str(e)))